test/vaadin-list-mixin.html

```diff
@@ -635,13 +637,31 @@
           expect(getComputedStyle(list.items[1]).getPropertyValue('display')).to.equal('none');
           expect(list.items[2].focused).to.be.true;
         });
+
         it('should move focus to next not hidden element on "arrow-up"', () => {
           expect(list.items[0].focused).to.be.true;
           expect(getComputedStyle(list.items[0]).getPropertyValue('display')).to.equal('block');
           arrowUp(list);
           expect(getComputedStyle(list.items[list.items.length - 1]).getPropertyValue('display')).to.equal('none');
           expect(list.items[list.items.length - 2].focused).to.be.true;
         });
+
+        it('should not set tabIndex=0 to hidden items, but the next one in the loop', () => {
+          list._focus(1);
```

Comment:
nit: Avoid using private APIs in the tests.

Reply:
fixed!

Reply:
Now the tests pass even without this change https://github.com/vaadin/vaadin-list-mixin/pull/64/files#diff-9e82cf3b8875402729200020ea93243bL212-R216

See if something is not working correctly without the change. If everything works, the change isn't needed; if something is broken, fix the test to check the right things.

Reply:
hi !
I tried to use external API in tests:
function arrowDown(target) {
      MockInteractions.keyDownOn(target, 40, [], 'ArrowDown');
    }
. In source code, I checked hidden item in _setFocusable(idx) -> _focus(idx) -> _onKeydown(event).
The test will be failed if without checking hidden items. and not work correctly and we type first letter, the hidden items can be focused and tab index will be = 0.

Reply:
I don't understand. The situation is still the same. The tests pass even without the change.

Reply:
I checked it and as you told me, I check the element is hidden in _setFocusable(idx) is not affected any. So I removed it.